### PR TITLE
Axisymmetric filters

### LINKED ISSUE #9: Generating Filters
**Translate the core functions which generate the wavelet filters from s2let (c) to s2wav (base python).**

Difficulty: 
- Medium

Background: 
- The wavelet transform works by "tiling" the harmonic domain with sets of filters, which can be designed for particular applications. Using filter functions (which must already be translated) we can essentially iteratively generate a vector of filters.

S2let File to translate: 
- can be found [here](https://github.com/astro-informatics/s2let/blob/main/src/main/c/s2let_tiling.c) but **don't** include lines 41-110.

S2Wav File location: 
- should be put in the main package directory [here](https://github.com/astro-informatics/s2wav/tree/main/s2wav)

Notes: 
- Remember that in python we don't need to allocate memory, so we can just start with standard numpy arrays of the correct dimension and ignore commands like "calloc" etc.

### PR DESCRIPTION
**Pull request by alicjapolanska**
Calculate wavelet kernels and scaling kernel in harmonic space.

Closes #9

### CODE REVIEW

File: s2wav/axisym_filters.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+def k_lam(L:int, lam:float, quad_iters: int=300) -> float:
+    """Compute function k_lam defined in equation (12) in [1] used to compute scaling function generating function and
+        wavelet generating function.
+
+    Args:
+        L (int): Harmonic band-limit.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        quad_iters (int): Total number of iterations for quadrature integration. Defaults to 300.
+    Returns:
+        np.ndarray: value of k_lam computed for values between 1/lam and 1, parametrised by l as required to compute
```
Comment (JessWhitney) [inline s2wav/axisym_filters.py:16]:
```suggestion
        (np.ndarray): value of k_lam computed for values between 1/lam and 1, parametrised by l as required to compute 
```

Comment (JessWhitney) [reply s2wav/axisym_filters.py:16]: Think the data type needs to be in brackets for it to be recognized correctly in the google doc style.

File: s2wav/axisym_filters.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+def filters_axisym(L: int, lam:float, J_min: int) -> Tuple[np.ndarray]:
+    """Computes wavelet kernels and scaling kernel in harmonic space, according to equations (15), (16) in [1]
+        without the normalisation factor.
+    
+    Args:
+        L (int): Harmonic band-limit.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        J_min (int): First wavelet scale used.
+    Returns:
+        Tuple[np.ndarray]: Unnormalised wavelet kernels and scaling kernel in harmonic space.
```
Comment (JessWhitney) [inline s2wav/axisym_filters.py:51]: I'd recommend adding a couple of extra lines below defining what phi and psi are.

**Review by JessWhitney — COMMENTED** [review]
Looks good, just a couple formatting stuff I spotted.

File: s2wav/axisym_filters.py (hunk 2 above)
Comment (CosmoMatt) [reply s2wav/axisym_filters.py]: I've added a few words to describe these things.

File: s2wav/axisym_filters.py (hunk 1 above)
Comment (CosmoMatt) [reply s2wav/axisym_filters.py]: I'm not 100% certain this is mandatory, but we should be consistent throughout the code, and this is the convention used elsewhere :) I've updated appropriately.

**Review by CosmoMatt — APPROVED** [review]
All looks great @alicjapolanska ! I've just address @JessWhitney's comments and added some details to the docstrings.

**Pull request merged**